### PR TITLE
Fixed doubleclick and autorender bugs, extended UI

### PR DESCRIPTION
**Pull request by flip-md**
Hey @graydon,

+ Added settings, grouping options, sorting & saving of state
+ Modified README.MD (removed bug description, added overview video)
+ bumped version to 0.3.0

.. this one got a bit bigger, let me know what you think!

### CODE REVIEW

**Comment by graydon** [comment]
@flip-md wow thanks! I'll need a little time to review (and actually .. set back up my obsidian development workflow to the point that I remember how to even check this) but will try to merge & publish shortly!

File: controller/fileItem.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { Config } from '../service/config';
+import { ICONS, NAVIGATION_TAB } from '../constants';
+import { Sortable } from './navigationItem';
+
+export default class DanglingFileItem extends Sortable {
+    constructor(app: App) {
+        super(ICONS.FILE_ICON, NAVIGATION_TAB.FILE, 'File', app);
+    }
+
+    getDanglingLinks(): Map<string, Dangle[]> {
```
Comment (graydon) [inline controller/fileItem.ts:12]: Hmm, this code is reproduced in all 3 controllers -- could you factor it out?

**Review by graydon — CHANGES_REQUESTED** [review]
Looks fairly good! Only one obvious change stands out -- it'd be nice not to have the dangling-link calculation itself in one place rather than triplicated. Unless I'm misunderstanding.

Otherwise happy to take it, tested locally and it works (very nicely), thanks!

File: controller/fileItem.ts (hunk 1 above)
Comment (flip-md) [reply controller/fileItem.ts:16]: Factored it out. There is still some custom logic in each controller as the grouping is handled differently per controller.

**Comment by flip-md** [comment]
Addressed your review and added a bugfix for expansion of the summary. Thank you!

**Review by graydon — APPROVED** [review]
Great, thanks!

**Pull request merged**